platform/osx/detect.py

```diff
@@ -115,6 +115,9 @@ def configure(env):
         env.Append(LINKFLAGS=["-isysroot", "$MACOS_SDK_PATH"])
 
     else:  # osxcross build
+        env.Append(CCFLAGS=["-mmacosx-version-min=10.12"])
```

Comment:
I guess we could mirror what is done for the Xcode path with support for `arm64` (osxcross doesn't support it yet, but when it does, I guess the same options will be needed). Or directly move lines 90-97 before the compiler-specific branches.

Reply:
For OSXCross there's also this line that should be branched for ARM64.

https://github.com/godotengine/godot/blob/aada93afc0ecc699440805ad2574d3e94b7c0abd/platform/osx/detect.py#L119